File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        
        n_dict = {}
        for n in nums:    
            if n in n_dict:
                n_dict[n] += 1
            else:
                n_dict[n] = 1
                
        for n in n_dict:
            if n_dict[n] == 1:
                return n

File: 0136-single-number/0136-single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
        map<int, int> n_dict;
        for (int i = 0; i < nums.size(); i++) {
            if (n_dict.find(nums[i]) != n_dict.end())
                n_dict[nums[i]]++;
            else
                n_dict[nums[i]] = 1;
        }

        for (auto i = n_dict.begin(); i != n_dict.end(); i++) 
            if (i->second == 1)
                return i->first;
        return 0;
    }
};